tools/functional-tester/etcd-runner/command/election_command.go

```diff
@@ -65,18 +71,30 @@ func runElectionFunc(cmd *cobra.Command, args []string) {
 				break
 			}
 		}
-		e := concurrency.NewElection(s, "electors")
 
-		rcs[i].acquire = func() error {
-			<-releasec
+		e := concurrency.NewElection(s, election)
+		rcs[i].acquire = func() (err error) {
 			ctx, cancel := context.WithCancel(context.Background())
+			donec := make(chan struct{})
 			go func() {
-				if ol, ok := <-e.Observe(ctx); ok {
+				defer close(donec)
+				ol, ok := <-e.Observe(ctx)
```

Comment:
In the event of Observe on leader whose rev < compacted rev, Observe will be closed due to the underlying watch ch closes on compacted rev. Retrying on observe will be the same behavior.  However, Leader() will return leader information base on whether the leader key exists or not. That's why I spin on Leader(). Not sure if that's the best approach tho.